### DISCUSSION TITLE
過長的Webpack compile，如何配置Babel去忽略 node_modules?

### DISCUSSION THREAD

**Discussion by ghost** [umijs/umi#12386] [General]
現在build webpack的時候因為babel長期卡在node modules裡了，我查了解決方法好像是配置 babel loader把node_modules給exclude掉，但在Umi 裡找不到關於babel loader的config。請問如何配置 babel loader或有甚麼方法改善？
```
[BABEL] Note: The code generator has deoptimised the styling of /app/node_modules/jodit/build/jodit.min.js as it exceeds the max of 500KB.
✔ Webpack: Compiled successfully in 2.64m
```
Umi version: 3

**Comment by ghost** [comment]
找到類似的discussion了： https://github.com/umijs/umi/discussions/7378

**Comment by fz6m** [comment]
umi 3 有一个 [nodeModulesTransform](https://v3.umijs.org/config#nodemodulestransform) 选项。

umi 3 不维护了，建议升级为 umi 4
